packages/mds-cache/index.ts

```diff
@@ -122,6 +131,24 @@ async function hread(suffix: string, device_id: UUID): Promise<CachedItem> {
   throw new Error(`${suffix} for ${device_id} not found`)
 }
 
+async function addGeospatialHash(key: string, coordinates: [number, number]) {
+  const client = await getClient()
+  const [lat, lng] = coordinates
+  const res = await client.geoadd('locations', lng, lat, key)
+  return res
+}
+
+async function getEventsInBbox(bbox: BoundingBox) {
+  const client = await getClient()
+  const [pt1, pt2] = bbox
+  const [lng, lat] = [(pt1[0] + pt2[0]) / 2, (pt1[1] + pt2[1]) / 2]
+  const [radius, unit] = distance
```

Comment:
We have our own haversine function in utils ... perhaps we should remove it and use this everywhere.